core/core-funcs.el

```diff
@@ -36,12 +36,18 @@
   (run-hooks 'text-mode-hook))
 
 (defun spacemacs//get-package-directory (pkg)
-  "Return the directory of PKG. Return nil if not found."
-  (let ((elpa-dir (file-name-as-directory package-user-dir)))
-    (when (file-exists-p elpa-dir)
-      (let* ((pkg-match (concat (symbol-name pkg) "-[0-9]+"))
-             (dir (car (directory-files elpa-dir 'full pkg-match))))
-        (when dir (file-name-as-directory dir))))))
+  "Return the installed directory of PKG. Return `nil' if not found."
+  (let ((directories (cons package-user-dir
```

Comment:
Why not change configuration-layer//get-package-directory?
It never occurred to me.  In my testing just applying the change in this PR was sufficient in that
spacemacs honored the packages installed under site-lisp and so it did not install them under
package-user-dir.